### PR TITLE
FB and slideshow embedded media fix

### PR DESCRIPTION
**Pull request by mparfeniuk**
I'd like to keep amp-helper-functions.php and class-amp-autoloader.php clear but for now I included our custom classes here

### CODE REVIEW

**Comment by nkranes** [comment]
@mparfeniuk, In general, can we keep our code as separate from the main codebase as possible to allow for future merges from the Automattic repo?

**Comment by mparfeniuk** [comment]
I'm searching for approach to work around the issue

**Comment by nkranes** [comment]
Thanks @mparfeniuk. Or at least minimize it.

**Comment by mparfeniuk** [comment]
I did refactoring,  minimizing intervention to wp code to 1 line. I removed '!' symbol from the filepath. I suppose it's a bug from wp developer but not sure

File: includes/amp-helper-functions.php [hunk 1]
```diff
@@ -583,26 +583,26 @@ function amp_get_content_embed_handlers( $post = null ) {
...
+			'AMP_Pinterest_Embed_Handler'             => array(),
+			'AMP_Playlist_Embed_Handler'              => array(),
+			'AMP_Reddit_Embed_Handler'                => array(),
+			'AMP_Tumblr_Embed_Handler'                => array(),
+			'AMP_Gallery_Embed_Handler'               => array(),
+			'AMP_Gfycat_Embed_Handler'                => array(),
+			'AMP_Hulu_Embed_Handler'                  => array(),
+			'AMP_Imgur_Embed_Handler'                 => array(),
+			'WPCOM_AMP_Polldaddy_Embed'               => array()
 		),
```
Comment (mattfrankly) [inline includes/amp-helper-functions.php]: Can you please remove the tab/comma changes here to revert back to the original file.?

**Review by mattfrankly — CHANGES_REQUESTED** [review]

File: includes/class-amp-autoloader.php [hunk 2]
```diff
@@ -161,6 +161,7 @@ public static function register() {
 	 * @param string $filepath   Absolute filepath to class file, including .php extension.
 	 */
 	public static function register_autoload_class( $class_name, $filepath ) {
-		self::$_classmap[ $class_name ] = '!' . $filepath;
+		//self::$_classmap[ $class_name ] = '!' . $filepath;
+		self::$_classmap[ $class_name ] = $filepath; // Removing '!' from the filepath
```
Comment (mattfrankly) [inline includes/class-amp-autoloader.php]: @mparfeniuk , why doesn't this work as originally coded?

File: includes/embeds/frankly-amp-embeds/class-frankly-amp-slideshow-embed.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	public function register_embed() {
+		return true;
+	}
+
+	public function unregister_embed() {
+		return true;
+	}
+
+    /**
+	 * Sanitized <div class="fb-video" data-href=> tags to <amp-facebook>.
```
Comment (mattfrankly) [inline includes/embeds/frankly-amp-embeds/class-frankly-amp-slideshow-embed.php]: please update the comment to reflect the proper class name.

File: includes/embeds/frankly-amp-embeds/class-frankly-amp-facebook-embed.php [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+				return 'post';
+			}
+
+        } elseif (false !== strpos($string, 'video')) {
+            return 'video';
+        }
+    }
+
+    /**
+     * Sanitized <div class="fb-video" data-href=> tags to <amp-facebook>.
```
Comment (mattfrankly) [inline includes/embeds/frankly-amp-embeds/class-frankly-amp-facebook-embed.php]: please update this comment.

**Review by mattfrankly — CHANGES_REQUESTED** [review]

File: includes/amp-helper-functions.php (hunk 1 above)
Comment (mparfeniuk) [reply includes/amp-helper-functions.php]: ohh yes

**Pull request merged**